pegasus/sites.v3/code.org/views/learn_carousels.haml

```diff
@@ -1,50 +1,51 @@
+  all_categories['learn'] << {tag:"Teacherled", name: I18n.t(:teach_led_lesson_plans)}
+  
+  all_categories['beyond'] = []
+  all_categories['beyond'] << {tag:"Beginner", name: I18n.t(:carousel_heading_beyond_beginners)}
+  all_categories['beyond'] << {tag:"JavaScript", name: I18n.t(:carousel_heading_beyond_javascript)}
+  all_categories['beyond'] << {tag:"Apps", name: I18n.t(:carousel_heading_beyond_devices)}
+  all_categories['beyond'] << {tag:"OtherLang", name: I18n.t(:carousel_heading_beyond_languages)}
+  all_categories['beyond'] << {tag:"MakeApps", name: I18n.t(:carousel_heading_apps_games)}
+  all_categories['beyond'] << {tag:"Univ", name: I18n.t(:carousel_heading_university)}
+  all_categories['beyond'] << {tag:"Web", name: I18n.t(:carousel_heading_webpages)}
+  all_categories['beyond'] << {tag:"Robot", name: I18n.t(:carousel_heading_robots)}
+  all_categories['beyond'] << {tag:"IDE", name: I18n.t(:carousel_heading_ide)}
+  
+  all_categories['elementary'] = []
+  all_categories['elementary'] << {tag:"Desktop", name: I18n.t(:carousel_heading_desktop)}
+  all_categories['elementary'] << {tag:"MobTabElem", name: I18n.t(:carousel_heading_mobtabelem)}
+  all_categories['elementary'] << {tag:"Prereader", name: I18n.t(:carousel_heading_prereader)}
+  all_categories['elementary'] << {tag:"BeyondBlocks", name: I18n.t(:carousel_heading_beyondblocks)}
+  all_categories['elementary'] << {tag:"NoInternet", name: I18n.t(:carousel_heading_nointernet)}
+  all_categories['elementary'] << {tag:"RobotsElem", name: I18n.t(:carousel_heading_robots)}
+  
+  all_categories['middle_high'] = []
+  all_categories['middle_high'] << {tag:"JavaScript", name: I18n.t(:carousel_heading_beyond_javascript)}
+  all_categories['middle_high'] << {tag:"Beginner", name: I18n.t(:carousel_heading_beyond_beginners)}
+  all_categories['middle_high'] << {tag:"Web", name: I18n.t(:carousel_heading_webpages)}
+  all_categories['middle_high'] << {tag:"OtherLang", name: I18n.t(:carousel_heading_beyond_languages)}
+  all_categories['middle_high'] << {tag:"MobTabMid", name: I18n.t(:carousel_heading_beyond_devices)}
+  all_categories['middle_high'] << {tag:"RobotsMid", name: I18n.t(:carousel_heading_robots)}
+  
+  all_categories['university'] = []
+  all_categories['university'] << {tag:"Univ", name: I18n.t(:carousel_heading_university)}
+  all_categories['university'] << {tag:"Web", name: I18n.t(:carousel_heading_webpages)}
+  all_categories['university'] << {tag:"JavaScript", name: I18n.t(:carousel_heading_beyond_javascript)}
+  all_categories['university'] << {tag:"OtherLangUniv", name: I18n.t(:carousel_heading_beyond_languages)}
+  all_categories['university'] << {tag:"RobotsUniv", name: I18n.t(:carousel_heading_robots)}
 
 %div{style: "clear:both"}
 
--tutorials = Tutorials.new(variation=='learn' ? :tutorials : :beyond_tutorials)
+- tutorials = Tutorials.new(variation=='learn' ? :tutorials : :beyond_tutorials)
```

Comment:
This could just get moved up into the `:ruby` block above.  Also the line of ruby below.